Bitbucket.Authentication/Authentication.cs

```diff
@@ -281,7 +313,7 @@ public static BaseAuthentication GetAuthentication(TargetUri targetUri, ICredent
         /// <returns>a valid instance of <see cref="Credential"/> or null</returns>
         public async Task<Credential> InteractiveLogon(TargetUri targetUri, string username)
         {
-            if (string.IsNullOrWhiteSpace(username))
+            if (string.IsNullOrWhiteSpace(username) || TargetUriContainsUsername(targetUri))
```

Comment:
Would the `ContainsUsername` be better served as a method or property on `TargetUri` rather than a `static` in `BitBucket.Authentication` ?